05/Task05.js

```diff
@@ -1,8 +1,52 @@
-import React from 'react';
+import React, {useState, useEffect} from 'react';
+import Shop from '../src/components/Shop';
+import { useParams, Switch, Route, useHistory, Redirect } from 'react-router-dom';
+import products from "../src/products.json"
+import ShopWithSortParams from '../src/components/ShopWithSortParams';
+
 
 const Task05 = () => {
+    const history = useHistory();
+
+    const [state, setState] = useState({
+        minPrice: '',
+        maxPrice: '',
+        searchTerm: '',
+    })
+
+    const handleChange = event => {
+        const {name, value} = event.target;
+        setState({
+            ...state,
+            [name]: value
+        })
+    }
+    useEffect(()=>{
+        const {minPrice, maxPrice, searchTerm} = state
+        if(minPrice !== '' || maxPrice !== '' || searchTerm !== ''){
+            history.push(
+                `/task05/${minPrice}-${maxPrice}-${searchTerm}`
+            )
+        }else{
+            history.replace('/task05')
+        }
+    }, [state])
+        
     return (
-        <h1>Task05</h1>
+        <>
+            <h1>Task05</h1>
+            <form>
+                <input onChange={handleChange} value={state.minPrice} name="minPrice" type="number" placeholder='cena minimalna'></input>
+                <input onChange={handleChange} value={state.maxPrice} name="maxPrice" type="number" placeholder='cena maksymalna'></input>
+                <input onChange={handleChange} value={state.searchTerm} name="searchTerm" type="text" placeholder='fraza'></input>
+            </form>
+            <Route exact path={`/task05`}>
+                <Shop products={products}></Shop>
+             </Route>
+             <Route path='/task05/:min?-:max?-:term?'>
```

Comment:
Prosto nie będzie, ponieważ mamy `min` oraz `max`, który jest liczbą i majać 12 nie wiemy czy 12 to max czy może 1 to min, a 2 to max. Dlatego zostały dodane myślniki. Przy takim rozwiązaniu wystarczy zrobić w ten sposób, że gdy dane są wysyłąne z formularza to przypisywać im domyślne wartości tj. `if(minEl.value === '') min = 0` lub `if(macEl.value === '') max = Number.MAX_VALUE`